compiler/src/iree/compiler/Dialect/Stream/Transforms/SpecializeEncodings.cpp

```diff
@@ -723,6 +727,24 @@ updateResultEncoding(RewriterBase &rewriter, OpTy op,
   return success();
 }
 
+/// Updates the source encoding and the result encoding of `op` with resolved
+/// layouts.
+static LogicalResult
+updateTensorEncodeOp(RewriterBase &rewriter, IREE::Stream::TensorEncodeOp op,
+                     const SetVector<Attribute> &layoutResolvers) {
+  if (failed(updateResultEncoding(rewriter, op, layoutResolvers))) {
+    return failure();
+  }
+  auto encodingType = dyn_cast<RankedTensorType>(op.getSourceEncoding());
+  Type newEncodingType =
+      getTypeWithResolvedEncodingLayouts(encodingType, layoutResolvers);
+  if (!newEncodingType) {
+    return op.emitOpError("failed to resolve recognized layout");
+  }
+  rewriter.modifyOpInPlace(op, [&] { op.setSourceEncoding(newEncodingType); });
+  return success();
```

Comment:
Not sure whether you see any further use cases for this, but for consistency with `updateResultEncoding`, you could also put this in a standalone `updateSourceEncoding` function?

Reply:
Good point, thanks!